### PR TITLE
Tenant settings/config update and access mechanism

### PR DESCRIPTION
**Pull request by shaangill025**
- redo of #589 which had to be reverted due to an import bug
- Commit [1a1507e] are changes for the per tenant settings
- Commit [6bae931] is basically cleanup from black formatter check

### CODE REVIEW

**Comment by loneil** [comment]
Test tennant for this PR
744d2212-b437-4c29-8495-f331a6b2cb30
6bcebf9e-9751-4fb9-be24-8c9d8f8cc3c2

Basic functionality looking good

**Comment by loneil** [comment]
@shaangill025 I could be missing, but what's the mechanism for the Innkeeper to **set** the allowance flags for each Tenant?

**Comment by loneil** [comment]
@esune can you confirm the business requirements we want to support by this feature are as follows (see upcoming Tenant UI concept)?

Given I'm an Innkeeper managing a Traction with several tenants I want to set a flag (or expand to multiple flags for multiple ledgers) for any of the specific Tenants:
image

image

Then each tenant is either allowed or not allowed to connect to an endorser and/or register DID:
image

**Comment by esune** [comment]
> @esune can you confirm the business requirements we want to support by this feature are as follows (see upcoming Tenant UI concept)?
> 
> Given I'm an Innkeeper managing a Traction with several tenants I want to set a flag (or expand to multiple flags for multiple ledgers) for any of the specific Tenants: image
> 
> image
> 
> Then each tenant is either allowed or not allowed to connect to an endorser and/or register DID: image

Yep, that sounds exactly like what I had in mind. 👍🏻

**Comment by shaangill025** [comment]
> @shaangill025 I could be missing, but what's the mechanism for the Innkeeper to **set** the allowance flags for each Tenant?

@loneil With the latest commits, I have updated `ReservationRecord` and added `connect_to_endorsers` & `create_public_did` attributes. Via tenant-ui, `/innkeeper/reservations/{reservation_id}/config` can be used to put these settings in the `ReservationRecord`, these will be applied when creating tenant sub-wallet upon reservation check-in. `/innkeeper/tenants/{tenant_id}/config` can be used to update these settings for tenant after reservation process [and sub-wallet creation].

Settings for `Innkeeper` via plugin config during startup remains same as before.

File: services/tenant-ui/frontend/src/store/tenantStore.ts [hunk 1]
```diff
@@ -196,6 +196,16 @@ export const useTenantStore = defineStore('tenant', () => {
...
+      console.log('> tenantStore.config');
+      await acapyApi
+        .getHttp(API_PATH.TENANT_CONFIG)
+        .then((res: any) => {
+          console.log(res);
+        })
+        .catch((err) => {
+          console.log(err);
+        });
+      console.log('< tenantStore.config');
```
Comment (loneil) [inline services/tenant-ui/frontend/src/store/tenantStore.ts:208]: This part can get taken out, I'll handle the loading in the store where needed

**Comment by esune** [comment]
@loneil I'll let you approve/merge the PR since you're working on the frontend part related to this change.

**Comment by loneil** [comment]
@esune It's not working at the moment, Tenant calls aren't resolving
image

and I'm not seeing the traction-tenant or traction-innkeeper categories in the Swagger any more https://pr-604-traction-acapy-admin-dev.apps.silver.devops.gov.bc.ca/api/doc

Think @shaangill025 mentioned he was still looking into it on friday

**Pull request closed without merging**